schema/me/conversation.js

```diff
@@ -173,12 +202,24 @@ export const ConversationFields = {
     type: connectionDefinitions({ nodeType: MessageType }).connectionType,
     description: "A connection for all messages in a single conversation",
     args: pageable(),
-    resolve: ({ messages }, options) => {
-      const impulseOptions = parseRelayOptions(options)
-      return connectionFromArraySlice(messages, options, {
-        arrayLength: messages.length,
-        sliceStart: impulseOptions.offset,
-      })
+    resolve: ({ id, from_email_address }, options, req, { rootValue: { accessToken } }) => {
+      const { page, size, offset } = parseRelayOptions(options)
+      const impulseParams = { page, size, conversation_id: id }
+      return gravity
+        .with(accessToken, { method: "POST" })("me/token", {
+          client_application_id: IMPULSE_APPLICATION_ID,
+        })
+        .then(data => {
+          return impulse
+            .with(data.token, { method: "GET" })(`message_details`, impulseParams)
+            .then(({ total_count, message_details }) => {
+              message_details["conversation_from_address"] = from_email_address
```

Comment:
This deserves some explanation.

We inject, as `conversation_from_address`, into each message payload, the _unmasked_ email of the originator of the conversation. This is presumably the user's email address.

Thus, we can tell on a _per-message_ basis (in https://github.com/artsy/metaphysics/pull/680/files#diff-e1fb60e2b771055ea7d7b51ceec6aa31R78), if the message is from a user or not (and can render their name, or the partner logo).

Believe it or not this is the best way we can currently check the `from_type` of a particular message...we'll most likely add something more structured that will let us determine this soon.